[EPM-3]상세보기 페이지

xloha opened the pull request:
* 셀 선택 시 상세보기 페이지로 이동

xloha committed:
[EPM-3]상세보기 페이지
* 셀 선택 시 상세보기 페이지로 이동

jangDta commented on EUNPPINGMALL/Controller/PopUpViewController.swift:
```diff
@@ -0,0 +1,60 @@
+    @IBOutlet weak var desc: UILabel!
+    @IBOutlet weak var heart: UIButton!
+    @IBAction func didTapHeart(_ sender: Any) {
+        if Product.dummyProductList[heart.tag].wish {
```
configure에서는 product를 바라보고 didTap에서는 Product.dummyProductList[heart.tag]를 보고 있네요
팝업VC가 바라보는 데이터는 product 하나여야 사이드 이펙트가 적을 것 같습니다

xloha replied:
넘어온 데이터인 product의 wish를 바라보고, 저장만 Product.dummyProductList[heart.tag]에 하도록 하였습니다.

jangDta commented on EUNPPINGMALL/Controller/PopUpViewController.swift:
```diff
@@ -0,0 +1,60 @@
+        self.price.text = p.price.commaRepresentation
+        self.desc.text = p.description
+
+        if let img = p.img {
```
취향의 차이지만 저라면 한줄로 짤거같아요
p.img 랑 p.wish

xloha replied:
수정하였습니다.

jangDta commented on EUNPPINGMALL/Controller/RecentViewController.swift:
```diff
@@ -15,6 +15,14 @@ class RecentViewController: UIViewController {
     }
     
+    func showProductPopUp(index: Int) {
+        guard let vc = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "PopUpViewController") as? PopUpViewController else { return }
```
enum을 마스터하신거같으니 storyboard도 Enum으로 빼볼가요
identifier도 저번 플젝때 넣엇던 거 가져다가 쓰시면 될거같습니다
이쯤이면 Eunum으로 레포 하나 파시죠🤓

xloha committed:
[EPM-3]상세보기 페이지
* 셀 선택 시 상세보기 페이지로 이동
* Layout 콘솔 로그 오류 해결
---
Extension
NSObject > reuseIdentifier추가 : reuseIdentifier를 반환함

Utility 폴더 추가
* ViewControlllerUtility : Storyboard이름과, viewController의 ID를 받아서 UIViewController return
* PopUpUtility : 띄울 UIViewController와 띄울 Product를 받아서 PopUp을 띄움

jangDta commented on EUNPPINGMALL/Controller/PopUpViewController.swift:
```diff
@@ -45,16 +46,8 @@ class PopUpViewController: UIViewController {
-        }
+        self.img.image = UIImage(named: p.img ?? "default")
+        
+        p.wish ? self.heart.setImage(UIImage(systemName: "heart.fill"), for: .normal) : self.heart.setImage(UIImage(systemName: "heart"), for: .normal)
```
```swift
self.heart.setImage(UIImage(systemName: p.wish ? "heart.fill" : "heart", for: .normal)
```

이렇게 할수도 있겠네용

xloha replied:
헉,,닙넵

jangDta commented on EUNPPINGMALL/Controller/RecentViewController.swift:
```diff
@@ -59,7 +51,7 @@ extension RecentViewController: UITableViewDataSource {
-        Product.dummyRecentList.insert(indexPath.row)
-        showProductPopUp(index: indexPath.row)
+        let index = Product.dummyRecentList[Product.dummyRecentList.index(Product.dummyRecentList.startIndex, offsetBy: indexPath.row)]
+        PopUpUtility.showProductPopUp(viewController: self, product: Product.dummyProductList[index])
```
여기 소스 2줄 설명좀 부탁드려요 ㅎ

xloha replied:
`Product.dummyRecentList`가 `Set` type이여서 원하는 index의 값을 가져오는 방법입니다.
`PopUpUtility`의 `showProductPopUp`함수를 부르면 인자로 전달된 `viewController`에 `product`를 팝업으로 띄우게 됩니다.

jangDta commented on EUNPPINGMALL/Utility/ViewControllerUtility.swift:
```diff
@@ -0,0 +1,15 @@
+
+import Foundation
+import UIKit
+
```
좋은방법이네요

제가 의도한건 이런 느낌이었어요
```swift
let vc = Storyboard.Main.instantiateViewController(withIdentifier: ProductViewController.reuseIdentifier)
```

jangDta replied:
아니면 이렇게도 할 수 있겠네요
```swift
struct Storyboard {
     static let main = UIStoryboard.init(name: "Main", bundle: nil)
     static let other = UIStoryboard.init(name: "Other", bundle: nil)

    struct vcIdentifier {
        static let product = ProductViewController.reuseIdentifier
        static let recent = RecentViewController.reuseIdentifier
        static let wishList = WishListViewController.reuseIdentifier
     }
}
```

```swift
let vc = Storyboard.main.instantiateViewcontroller(withIdentifier: Storyboard.vcIdentifier.product)
```

xloha replied:
두번째 방법이 제 취향이지만, 관리해야하는 view의 갯수가 많아질수록 해당 파일에 계속해서 추가해줘야한다는 점이 조금 걸리네요

jangDta replied:
해당 struct, enum을 구현하는 것은 소스의 가독성과 유지보수 측면, 소스내에서 ""를 쓰지 않아 오타를 예방하고 사이드이펙트를 없애기 위함일 것 같네용

jangDta commented on EUNPPINGMALL/Utility/PopUpUtility.swift:
```diff
@@ -0,0 +1,19 @@
+import Foundation
+import UIKit
+
+class PopUpUtility {
```
👍🏿
